src/Features/CSharp/Portable/UseIsNullCheck/CSharpUseIsNullCheckForCastAndEqualityOperatorCodeFixProvider.cs

```diff
@@ -25,14 +23,20 @@ internal class CSharpUseIsNullCheckForCastAndEqualityOperatorCodeFixProvider : S
         public override ImmutableArray<string> FixableDiagnosticIds
             => ImmutableArray.Create(IDEDiagnosticIds.UseIsNullCheckDiagnosticId);
 
+        private static bool IsSupportedDiagnostic(Diagnostic diagnostic)
+            => diagnostic.Properties[UseIsNullConstants.Kind] == UseIsNullConstants.CastAndEqualityKey;
```

Comment:
if it is 2 difference type of diagnostic that require 2 different fixer, is there a reason using same diagnostic id and use property bag to distinguish 2?

if it is going to use same diagnostic id, can 1 fixer take care of both cases so that fix all fixes both cases?

Reply:
if it is one option-concept, meaning people will think the diagnostic as the same one. then won't they expect fix all to fix both cases?

Reply:
also, if it is UI thing. UI can be 1 but that change could change both diagnostics IDs rather than merge them if diagnostic Ids are left as 2.

...

since it is now merged to 1. how the suppression and etc work? if one turns it off in ruleset or add suppression in global suppression file, both case get disabled I believe? so I feels like fix all should fix both cases... since suppression fix will affect both ones..

Reply:
not hard opinion though. just felt like it wasn't go all the way it is trying to go.

Reply:
> if it is one option-concept, meaning people will think the diagnostic as the same one. then won't they expect fix all to fix both cases?

Fix-all will fix both cases.  But we just have two fixers so we can have the individual cases handled through separate codepaths, instead of needing them to go through one.

Reply:
> also, if it is UI thing. UI can be 1 but that change could change both diagnostics IDs rather than merge them if diagnostic Ids are left as 2.

It could.  But that feels like it could be confusing.  For example, people would then have to suppress both diagnostics IDs in a file if they didn't want this feature working in a single file.

> so I feels like fix all should fix both cases...

fix-all does fix both cases.  This code here just ensures that each *fixer* is only responsible for some *portion* of the diagnostics.

Reply:
ah. alright then. I thought fix all doesn't fix both cases since it is 2 different fixers. I guess both fixer return same equivalent key or something. I thought the engine tied the key with 1 unique fixer. @mavasani